fanficfare/adapters/adapter_royalroadcom.py

```diff
@@ -166,12 +166,14 @@ def extractChapterUrlsAndMetadata(self):
 
 
         chapters = soup.find('table',{'id':'chapters'}).find('tbody')
-        tds = [tr.findAll('td')[0] for tr in chapters.findAll('tr')]
+        tds = [tr.findAll('td') for tr in chapters.findAll('tr')]
         # Links in the RR ToC page are in the normalized long form, so match is simpler than in normalize_chapterurl()
         chap_pattern_long = r"https?://(?:www\.)?royalroadl?\.com/fiction/\d+/[^/]+/chapter/(\d+)/[^/]+/?$"
-        for td in tds:
-            chapterUrl = 'https://' + self.getSiteDomain() + td.a['href']
-            if self.add_chapter(td.text, chapterUrl):
+        for chapter,date in tds:
+            chapterUrl = 'https://' + self.getSiteDomain() + chapter.a['href']
+            chapterDate = self.make_date(date)
+            if self.add_chapter(chapter.text, chapterUrl,
+                    {'date':chapterDate.strftime(self.getConfig("datechapter_format",self.getConfig("datePublished_format",self.dateformat)))}):
```

Comment:
Man this line is actually a nightmare to read, maybe I'll fix up the formatting a bit